src/helpers/amoc_xmpp_handlers.erl

```diff
@@ -57,12 +57,27 @@ stanza_handler(Pred, Action) ->
             end
     end.
 
+check_predicate(Pred, Stanza) when is_function(Pred, 1)->
```

Comment:
What is the benefit of this? For me it can hide important errors. If the predicate crashes, the client should crash as well. The fact that it crashes made me discover important bugs. In the new version of code I would simply miss them because they would be lost in the flood of logs.